src/visualization/ColorPalette/ColorMap.cc

```diff
@@ -0,0 +1,284 @@
+                }
+            }
+            ///< If its a new kmer, so we push it here
+            if (!matched)
+                kmerDeclarations.push_back(Sequences[i].substr(index*k, k));
+            matched = false;
+            index += 1;
+        }
+    }
+    differentKmers = kmerDeclarations;
+}
+
+/**
+ * This Method will make a vector of Colors in terms of a number, which defines the Length of the 
+ * length of the List. Then it will update our ColorList with it. In our case the argument 
+ * describes the number of Kmers we are looking for to map.
+ */
+void ColorMap::updateColorList(int numbOfKmer){
+    vector<Color> returnList; ///< the list i use to adjust this as the new List
+    Color placeholder; ///< A placeholder which is needed to programm with rgb values
+    /**
+     * In this part of the method i push Colors in the List which i searched in the
+     * web, so that they are well different.
+     */
+    placeholder.r = 255;
+    placeholder.g = 0;
+    placeholder.b = 0;
+    returnList.push_back(placeholder);
+    placeholder.g = 128;
+    returnList.push_back(placeholder);
+    placeholder.g = 255;
+    returnList.push_back(placeholder);
+    placeholder.r = 128;
+    returnList.push_back(placeholder);
+    placeholder.r = 0;
+    returnList.push_back(placeholder);
+    placeholder.b = 128;
+    returnList.push_back(placeholder);
+    placeholder.b = 255;
+    returnList.push_back(placeholder);
+    placeholder.g = 128;
+    returnList.push_back(placeholder);
+    placeholder.g = 0;
+    returnList.push_back(placeholder);
+    placeholder.r = 128;
+    returnList.push_back(placeholder);
+    placeholder.r = 255;
+    returnList.push_back(placeholder);
+    placeholder.b = 128;
+    returnList.push_back(placeholder);
+    placeholder.r = 128;
+    placeholder.g = 128;
+    placeholder.b = 128;
+    returnList.push_back(placeholder);
+    /**
+     * In this part of the method i push Colors in terms of a algorithm i made for myself.
+     * This algorithm is maybe not completed yet.
+     */
+    int index = 0; ///< i use this index to count about how much Colors are pushed.
+    int array[60];
+    for (int i=0;i<60;i++)
+        array[i] = 11 + i;
+    for (int l=59;l>=0;l--) {
+        for (int i=0;i<4;i++) {
+            for (int j=0;j<4;j++) {
+                for (int k=0;k<4;k++) {
+                    if (index==numbOfKmer) {
+                        l = -1;
+                        i = 4;
+                        j = 4;
+                        break;
+                    }
+                    index += 1;
+                    placeholder.r = (((i)*60)+array[l]);
+                    placeholder.g = (((j)*60)+array[l]);
+                    placeholder.b = (((k)*60)+array[l]);
+                    returnList.push_back(placeholder);
+                }
+            }
+        }
+    }
+    ColorList = returnList;
+}
+
+/**
+ * This method will map a single Kmer to a unique Color in terms of the 
```

Comment:
This should literally be a map, a standard C++ data type. Can be an `ordered_map` for example, that is more efficient and the code is easier to read.